staging/src/k8s.io/apimachinery/pkg/runtime/serializer/codec_factory.go

```diff
@@ -172,6 +183,12 @@ func (f CodecFactory) UniversalDeserializer() runtime.Decoder {
 	return f.universal
 }
 
+// UniversalStrictDeserializer is the same as UniversalDeserializer expect that it uses a strict Decoder
+// that catches unknown and duplicate fields.
```

Comment:
that (errors|throws an error) (on|for) unknown and duplicate fields

Reply:
also need to clarify:
* whether decoding continues in the presence of an unknown or duplicate field (is the decoded object usable?)
* what return value signifies an unknown/duplicate field
* how info about the bad fields is extracted from the returned error